### PR TITLE
[vLLM plugin] Fix 2D-mesh device-sampler garbage output (and add coherence check)

### LINKED ISSUE #4440: [vLLM plugin] Incorrect/random output generated with TP generative test
Tensor parallel generative model tests are generating incorrect/random output for 2D mesh. This issue is caused by sampling on device. These tests generate correct results with either 1D mesh or 2D mesh with cpu sampling. 

Output for last few nightly runs for `Qwen/Qwen3-32B` with 2D mesh are below 
[CI job 27 Apr](https://github.com/tenstorrent/tt-xla/actions/runs/25026871133/job/73299881403#step:19:439)
```
prompt: I like taking walks in the, 
output:  restartgp erect workplaceSignpart Unblock_REGSamplerlekestado                                           (info Htmldom ///building neighborhoods somewhat '''iness(())
```
[CI job 26 Apr](https://github.com/tenstorrent/tt-xla/actions/runs/24970760261/job/73131748597#step:19:439)
```
prompt: I like taking walks in the, 
output:  dou调 arrangementyieldysisyieldSignatureited bitstern(\ THE Need(isset Handle Parameter arts post Gretaphael Handle Penguin surplus@SpringBootTestystem language
```
[CI job 25 Apr](https://github.com/tenstorrent/tt-xla/actions/runs/24944092006/job/73042387279#step:19:439)
```
prompt: I like taking walks in the, 
output:  restart twittererase workflow winnerfectiontreeimiento Regionpartizophren Son класс PureComponent '''class.PORT\Requests_LE volunte Bool Personen Concf typically Sony
```
[CI job 24 Apr](https://github.com/tenstorrent/tt-xla/actions/runs/24917761316/job/72973189022#step:19:439)
```
prompt: I like taking walks in the, 
output:  dou--){
hen(Main.PORTsession Unblock Unblock reslek artscoctract]=>SQL Unblock.cHaving_play availabilityestureRecognizer shares Unblock ''' Char_OPTION
```

### PR DESCRIPTION
**Pull request by mmanzoorTT**
### Ticket
closes #4440 

### Problem description
- Tensor-parallel generative tests with 2D mesh + on-device sampling produced token-soup garbage (e.g. `'.presentientes winner actually! Bool volunte vertexlek language("");\n ReIGHlek...'`), while 1D mesh and 2D-mesh-with-CPU-sampling were both fine. Bisected to PR #3820 which gated the post-`compute_logits` replication-constraint to the 1D-mesh path only — the device sampler then operated on logits still sharded along the `model` axis.

- A second issue surfaced during the investigation: the bug had been landing as PASSED in CI because the tests asserted only that `llm.generate(...)` completed, not that its output was coherent. A small heuristic check is added so token-soup output now hard-fails.

### What's changed
- Update the condition in `compute_logits` to allow output tensor replication if ParallelLMHead layer is used for `compute_logits` for 2D mesh TP execution as the weights are sharded for ParallelLMHead layer.
- `tests/integrations/vllm_plugin/generative/conftest.py`: add `assert_output_coherent` — flags empty output, >3% non-Latin chars, or <10% common-English-stopword ratio as garbage.
- `tests/integrations/vllm_plugin/generative/test_tensor_parallel_generation.py`: call `assert_output_coherent(output_text)` after `llm.generate(...)` in the three TP generation tests (`_n300`, `_llmbox_small`, `_llmbox_large`).

### Checklist
- [X] New/Existing tests provide coverage for changes

### CODE REVIEW

**Review by kmabeeTT — APPROVED** [review]
Thx for workaround, will take a look at the issue https://github.com/tenstorrent/tt-xla/issues/4440!

**Comment by kmabeeTT** [comment]
Debugged issue and have fix. Since this missed nightly tonight, preference to skip this PR in leiu of proper fix to be merged tomorrow if you approve: https://github.com/tenstorrent/tt-xla/pull/4446

**Comment by mmanzoorTT** [comment]
More details about this issue can be found in #4446

**Review by kmabeeTT — APPROVED** [review]
Awesome, nice fix...thanks for including my checks and following up!
